src/Microsoft.ML.Data/DataView/DetectAnomalyBySrCnnBatchTransform.cs

```diff
@@ -0,0 +1,332 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.ML.Numeric;
+using Microsoft.ML.Runtime;
+
+namespace Microsoft.ML.Data.DataView
+{
+    public abstract class BatchTransformBase<TInput, TBatch> : IDataView
+    {
+        private protected sealed class Bindings : ColumnBindingsBase
```

Comment:
>Bindings [](start = 39, length = 8)

I think that since we are defining the `BatchTransformBase` class as an abstract base class, it may be better to put the `Bindings` in the deriving classes, in case one of the deriving class would not fit the scheme of one input column and one output column.